src/overlays/actors/ovl_En_Anubice/z_en_anubice.c

```diff
@@ -292,52 +299,53 @@ void EnAnubice_SetupDie(EnAnubice* this, GlobalContext* globalCtx) {
     this->animLastFrame = lastFrame;
     Animation_Change(&this->skelAnime, &gAnubiceFallDownAnim, 1.0f, 0.0f, lastFrame, ANIMMODE_ONCE, -20.0f);
 
-    this->unk_256 = false;
-    this->unk_258 = 0;
+    this->isNearWall = false;
+    this->fallTargetYaw = 0;
     this->deathTimer = 20;
     this->actor.velocity.x = this->actor.velocity.z = 0.0f;
     this->actor.gravity = -1.0f;
 
-    if (BgCheck_SphVsFirstPoly(&globalCtx->colCtx, &this->fireballPos, 70.0f)) {
-        this->unk_256 = true;
-        this->unk_258 = this->actor.shape.rot.x - 0x7F00;
+    if (BgCheck_SphVsFirstPoly(&globalCtx->colCtx, &this->headPos, 70.0f)) {
+        this->isNearWall = true;
+        this->fallTargetYaw = this->actor.shape.rot.x - 0x7F00;
     }
 
     this->actionFunc = EnAnubice_Die;
 }
 
 void EnAnubice_Die(EnAnubice* this, GlobalContext* globalCtx) {
     f32 curFrame;
-    f32 phi_f2;
-    Vec3f sp4C = { 0.0f, 0.0f, 0.0f };
+    f32 rotX;
+    Vec3f fireEffectInitialPos = { 0.0f, 0.0f, 0.0f };
     Vec3f fireEffectPos = { 0.0f, 0.0f, 0.0f };
     s32 pad;
 
     SkelAnime_Update(&this->skelAnime);
     Math_ApproachZeroF(&this->actor.shape.shadowScale, 0.4f, 0.25f);
 
-    if (this->unk_256) {
-        Math_SmoothStepToS(&this->actor.shape.rot.y, this->unk_258, 1, 10000, 0);
-        if (fabsf(this->actor.shape.rot.y - this->unk_258) < 100.0f) {
-            this->unk_256 = false;
+    // If near a wall, turn to avoid the tail falling through it during the death animation
+    if (this->isNearWall) {
+        Math_SmoothStepToS(&this->actor.shape.rot.y, this->fallTargetYaw, 1, 10000, 0);
```

Comment:
You think that's the explanation? 🤔 since `fallTargetYaw` is always set to the same thing instead of e.g. being relative to said wall (which btw doesn't have to be a wall, I think it could also be floor or ceiling if there was any nearby enough), it doesn't guarantee it will fall away from the wall

Reply:
its the intent at least. since the "rebound" yaw target is always `0 - 0x7F00` (since it uses x rot for god knows why), theres prob a scenario where the wall is already at that angle and it will turn into it. we can try to clarify that in an additional comment 

I think it would be more correct to say it tries to avoid the head from falling through the wall(poly) instead of the tail.

Reply:
It sounds like a sensible guess then, but should be worded as such to indicate being just a guess imo, like "this likely was intended to ... but is not functional"

Reply:
its functional as long as the wall does not have an angle of -0x7F00. I think all the context leading up to it makes it very clear that its what its supposed to do, its just bugged.

Theres no other reason why you would want to check for a nearby poly when dying, and changing the angle to fall to if its true, imo. its just they didnt do a sufficent job calculating which angle it should go

Reply:
I think a fair comment is 
```
// If near a wall, turn away from it while dying to avoid going through it.
// The implementation of this is bugged in the sense that the target angle is 
// hardcoded in practice. If the poly already has an angle of -0x7FFF, the expected behavior won't occur
```
you can choose to go into more detail about it being subtracted from x rotation which is always going to be 0, but its probably a bit too much to take in at once. dunno